iceoryx_posh/include/iceoryx_posh/mepoo/chunk_header.hpp

```diff
@@ -25,18 +26,45 @@ namespace iox
 {
 namespace mepoo
 {
+/// @brief Helper struct to use as default template parameter when no custom header is used
+struct NoCustomHeader
+{
+};
+
 /// @brief IMPORTANT the alignment MUST be 32 or less since all mempools are
 ///         32 byte aligned otherwise we get alignment problems!
 struct alignas(32) ChunkHeader
 {
-    ChunkHeader() noexcept;
+    using PayloadOffset_t = uint32_t;
+
+    /// @brief constructs and initializes a ChunkHeader
+    /// @param[in] chunkSize is the size of the chunk the ChunkHeader is constructed
+    /// @param[in] payloadSize is the size of the payload
+    /// @param[in] payloadAlignment is the alignment of the payload
+    /// @param[in] customHeaderSize is the size of the custom header; if no custom header is used, use 0
+    /// @param[in] customHeaderAlignment is the alignment for the custom header; if no custom header is used, use 1
+    ChunkHeader(const uint32_t chunkSize,
+                const uint32_t payloadSize,
+                const uint32_t payloadAlignment,
+                const uint32_t customHeaderSize,
+                const uint32_t customHeaderAlignment) noexcept;
```

Comment:
I think I will rename this to `userPayloadSize` ... `userHeaderAlignment` in a later PR. Then we can continue to configure our chunks like now with just the payload, which will be renamed to `chunkPayload`. With this we would have `chunk header` and `chunk payload` and the `chunk payload` could be split further into `user header` and `user payload`. Without `user header` and with the appropriate alignment `user payload` would have the same size as `chunk payload` which resembles the behavior before this changes.
```
+--------------+-------------------------------------------------------+
|              | chunk payload <- will be configured with the mempools |
| chunk header |+------------+-----------------------------------------+
| (fixed size) | user header | user payload                            |
+--------------+-------------+-----------------------------------------+
```

What do you think?